### ISSUE TITLE
Youtube stops working when used in background

### ISSUE THREAD

**Issue by Rtizer-9** [AdguardTeam/AdguardForAndroid#2604]
All the youtube variations from official app to vanced, newpipe and youtube website in browser stops working when used in BACKGROUND.

It's not just about the latest beta but almost every adguard I have used from starting had this error. This never happened in any other AdBlocker like adclear etc obviously because they do not block as much.

I think the culprit here is that I use Adguard spyware filter as well as easyprivacy and easylist both.

Whenever I look at filtering log when YouTube stops working (when screen is locked during background playback) the requests which are constantly being blocked are along the line of s.youtube.com/......... something 

The continuous blocking of those subdomains blocks YouTube from catching the video and it just stops there.

Is there something that can be done about this without giving up on Easyprivacy?

**Comment by zzebrum** [comment]
@R are you talking about picture in picture mode?

**Comment by Rtizer-9** [comment]
I'm talking about normal playback.... completely normal playback from Youtube or any other alternative WITH background playback i.e. either screen is locked or app is working in background.

That's what I want without any interruptions.

But after screen is locked, few moments after that the app stops caching and the song stops playing because of heavy adguard filtering.

It doesn't used to happen in adclear obviously because their filters are not so exhaustive.

**Comment by ameshkov** [comment]
Try adding `@@||s.youtube.com` to the user filter in AdGuard settings. Does it change anything for you?

**Comment by Rtizer-9** [comment]
I recently installed a custom rom based on nougat and I don't what custom setting is now applying differently, my YouTube experience has been completely changed.

Its just so smooth now. Not a single, I repeat single stuck error 502 either in comments or the video itself.

Apart from adguard the only Adblocking software I use is minminguard xposed module which only blocks in app ads (from sdk) and also has url blocking (only for those sdk networks).

I recently looked into minminguard and yes it was blocking mopubgms in my YouTube Vanced.

I'm sorry but I don't remember whether it was whitelisted like many other apps in my previous rom or not.
But now I have whitelisted it, as mopub is already blocked by adguard.

One strange thing I have observed is that on marshmallow I used to see a lot of heavy blocking from adguard filter, easyprivacy and dan pollock filters in s.youtube.com/.... "Blah blah ....my phone config.... blah blah"....

BUT now there are hardly any entries with s.youtube.com and the only occasional blocks are now from quic protocol and pagesyndication.

Earlier s.youtube.com was not whitelisted (well not this exact address but something extra was added after s.youtube.com in the whitelist).

So I checked if something changed in your official fiters and yes 5 months before you added that whitelist.

Now I'm confused why it never came into my filtering log and is only applying now.

Well I don't know the exact difference between 
s.youtube.com^ and
s.youtube.com^|
but I'm completely sure I never saw an entry with that whitelist even once.

Hope that helps you in some way and yes I think my problem is completely solved now.

I recall seeing different entries being blocked in ublock origin in chrome and Firefox for YouTube and many websites so its a proof ad networks injects themselves differently in different scenarios.

Well I didn't changed the user agent in my adguard android just so that you know and above observation is from desktop.
